core/corehttp/gateway.go

```diff
@@ -12,8 +12,7 @@ func GatewayOption(writable bool) ServeOption {
 		if err != nil {
 			return err
 		}
-		mux.Handle("/ipfs/", gateway)
-		mux.Handle("/ipns/", gateway)
+		mux.Handle("/", gateway)
```

Comment:
Here, I think we want to keep using the muxer. We're going towards:

```go
mux.Handle("/ipfs/", ipfsHandler)
mux.Handle("/ipns/", ipnsHandler)  // ipnsHandler is sort of middleware that uses ipfsHandler
mux.Handle("/", base)
```
where `baseHandler` can be mux between:

```go
base.Handle("/webui", redirectToWebUIHandler)
base.Handle("/", rootHandler) // give the user some information about IPFS + links (e.g. to WebUI, ipfs.io, etc)
// base should handle 404s with custom error pages.
```

cc @mappum